zebra/zebra_vrf.c

```diff
@@ -256,14 +259,12 @@ static int zebra_vrf_delete(struct vrf *vrf)
 
 	/* release allocated memory */
 	for (afi = AFI_IP; afi <= AFI_IP6; afi++) {
-		void *table_info;
-
 		for (safi = SAFI_UNICAST; safi <= SAFI_MULTICAST; safi++) {
 			table = zvrf->table[afi][safi];
 			if (table) {
-				table_info = route_table_get_info(table);
-				route_table_finish(table);
-				XFREE(MTYPE_RIB_TABLE_INFO, table_info);
+				zebra_router_release_table(zvrf, zvrf->table_id,
```

Comment:
If I'm reading the code correctly, `zebra_router_release_table()` is being called by both the VRF `disable` and `delete` callbacks, but `zvrf->table` is initialized in the VRF `create` callback only. So, if a VRF is disabled and then re-enabled, can't we have problems?

Reply:
yeah good point, I'll fix that.

Reply:
actually just double checked, table creation is done in enable, which seems ok to me.